fleetshard/pkg/central/reconciler/reconciler.go

```diff
@@ -385,6 +411,65 @@ func (r *CentralReconciler) reconcileCentralDBConfig(ctx context.Context, remote
 	return nil
 }
 
+func getAuditLogNotifierConfig(
+	auditLoggingConfig config.AuditLogging,
+	namespace string,
+) *declarativeconfig.Notifier {
+	return &declarativeconfig.Notifier{
+		Name: auditLogNotifierName,
+		GenericConfig: &declarativeconfig.GenericConfig{
+			Endpoint:            auditLoggingConfig.Endpoint(true),
+			SkipTLSVerify:       auditLoggingConfig.SkipTLSVerify,
+			AuditLoggingEnabled: true,
+			ExtraFields: []declarativeconfig.KeyValuePair{
+				{
+					Key:   auditLogTenantIDKey,
+					Value: namespace,
+				},
+			},
+		},
+	}
+}
+
+func (r *CentralReconciler) configureAuditLogNotifier(secret *corev1.Secret, namespace string) error {
+	if secret.Data == nil {
+		secret.Data = make(map[string][]byte)
+	}
+	auditLogNotifierConfig := getAuditLogNotifierConfig(
+		r.auditLogging,
+		namespace,
+	)
+	encodedNotifierConfig, marshalErr := yaml.Marshal(auditLogNotifierConfig)
+	if marshalErr != nil {
+		return fmt.Errorf("marshaling audit log notifier configuration: %w", marshalErr)
+	}
+	secret.Data[auditLogNotifierKey] = encodedNotifierConfig
+	return nil
+}
+
+func (r *CentralReconciler) reconcileDeclarativeConfigurationData(ctx context.Context, remoteCentral *private.ManagedCentral) error {
+	if !r.auditLogging.Enabled {
+		return nil
+	}
+	namespace := remoteCentral.Metadata.Namespace
+	exists, err := r.checkSecretExists(ctx, namespace, sensibleDeclarativeConfigSecretName)
+	if err != nil {
+		return err
+	}
+	if exists {
+		return nil
+	}
```

Comment:
super nit: can be simplified to
```suggestion
	if err != nil || exists {
		return err
	}
```